### DISCUSSION TITLE
rename master to main

### DISCUSSION THREAD

**Discussion by mikofski** [pvlib/pvlib-python#1584] [Ideas]
This has been a long time coming, but now that we are growing our ranks I think it's time to prioritize. GitHub has made it slightly easier, see [this post](https://github.com/github/renaming#renaming-existing-branches) - they will redirect pull requests and and links, but most contributors will still need to update their local environments.

More info in the [GitHub docs on renaming a branch](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/renaming-a-branch). In particular most user will need to follow [these steps](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/renaming-a-branch#updating-a-local-clone-after-a-branch-name-changes):
```bash
$ git branch -m OLD-BRANCH-NAME NEW-BRANCH-NAME
$ git fetch origin
$ git branch -u origin/NEW-BRANCH-NAME NEW-BRANCH-NAME
$ git remote set-head origin -a
$ git remote prune origin
```

If @pvlib/pvlib-core all agree then I propose we make this update before the end of 2022. Also please let me know if you think this discussion should be posted to the google group as well.

**Comment by AdamRJensen** [comment]
Seems like it has to happen sooner or later, so we might as well get it done!

**Comment by mikofski** [comment]
Maybe we should we put something temporarily in the developer docs, referencing GH?

**Comment by adriesse** [comment]
I don't think it *has* to happen sooner or later--it's a choice.  Anyway, I'm fine with it and have no opinion on the timing.

**Comment by cwhanse** [comment]
If we do this renaming, let's do it as the first action after the next release. I'm guessing that renaming will complicate any abandoned or stalled PRs, so we should complete or close those that have any future.

**Reply by mikofski** [reply]
As Will notes below, the only complication this will cause is for users' _local_ environments if they have cloned _master_ then they would need to follow the steps to rename it to _main_. Otherwise GH has made it painless and will redirect all links including PR's, issues, and discussions.

**Comment by wholmgren** [comment]
Looks like redirects should just work so my vote is to do it today. Failing that, make an issue and tag it with the next release milestone so we don't forget.

**Comment by mikofski** [comment]
ok it's done see #1587

**Comment by mikofski** [comment]
Steps to update your local environments and forks:
1. rename your default branch to "main" on GitHub for your personal fork. For me this is https://github.com/mikofski/pvlib-python/
2. run the bash commands to rename "master" in your local environment, track upstream, reset head, and prune any stale branches

       git branch -m master main
       git fetch upstream  # use the remote that points to https://github.com/pvlib/pvlib-python
       git fetch upstream
       git branch -u upstream/main main  # I don't think this line is actually necessary
       git remote set-head upstream -a
       git remote prune upstream

3. now repeat for remote that points to your fork

       git fetch origin  # use the remote that points to your GitHub fork
       git fetch origin
       git branch -u origin/main main  # I prefer main to track my personal fork, not upstream
       git remote set-head origin -a
       git remote prune origin
       git gc

4. repeat for any remotes that point to others' forks after they've renamed their default branch online at GitHub.

QED

**Reply by adriesse** [reply]
I did it!

**Comment by wholmgren** [comment]
We should make this change for `pvlib/pvanalytics` as well. Any objection to me making that change today @cwhanse @kanderso-nrel @kperrynrel? 

`pvlib/twoaxistracking` already uses `main`. Might as well make the change in `pvlib/pvmodules` in case that ever goes forward.

**Reply by cwhanse** [reply]
No objection, please do both. @abhisheksparikh FYI.

**Reply by wholmgren** [reply]
done!

**Reply by kandersolar** [reply]
FYI, ReadTheDocs seems to have not noticed this change, causing the `latest` docs to no longer be updated.  I've updated the configurations and triggered rebuilds of `latest` for both `pvlib/pvlib-python` and `pvlib/pvanalytics`.

More issues fixed in #1605

**Reply by mikofski** [reply]
nice catch! I hope no other external services are depending on manual input from a human to tell it what branch to pull from our repo
